src/loaders/utils/GetTakeoverDetails.ts

```diff
@@ -0,0 +1,96 @@
+import { isServerSide } from 'utils/getEnv';
+import { getArticleVideoServiceBaseUrl } from 'utils/getPandoraDetails';
+import { trackYoutubeTakeoverDetails, YoutubePlayerTrackingProps } from 'youtube/players/shared/youtubeTrackingEvents';
+import { YoutubeTakeOverDetails } from 'youtube/types';
+import { getMediaWikiConfigDetails } from 'loaders/utils/GetMediaWikiConfigDetails';
+import { VimeoTakeOverDetails } from 'vimeo/types';
+
+function getTakeoverUrl(wikiId: string): string {
+	const articleVideoBaseUrl = getArticleVideoServiceBaseUrl();
+	if (articleVideoBaseUrl.length === 0) {
+		return articleVideoBaseUrl;
+	}
+	if (!wikiId) {
+		return '';
+	}
+	return `${articleVideoBaseUrl}takeover/v1/takeover-mappings/${wikiId}`;
+}
+
+export async function getTakeoverDetails({ deviceType }: YoutubePlayerTrackingProps) {
+	const { wikiId, isTier3Wiki } = getMediaWikiConfigDetails();
+	if (!wikiId || isTier3Wiki) {
+		return null;
+	}
+
+	const response = await fetch(getTakeoverUrl(wikiId));
+	const dataArray = await response.json();
+	if (dataArray.length === 0) {
+		return null;
+	}
+
+	if (dataArray[0].youtube_take_over) {
+		const youtubeDetails: YoutubeTakeOverDetails = { isYoutubeTakeover: false };
```

Comment:
Can we clean up the typing a bit? I think we don't need separate types for YouTube and Vimeo as both have videoId and type flag. I think we could have
```
interface TakeoverDetails {
  videoId: string;
  type: 'youtube' | 'vimeo';
}

Reply:
this way we could avoid a lot of code duplication here

Reply:
Good idea, let me clean this up

Reply:
also, I think these `eligibleFor*Takeover` are redundant and we can just have a check for videoId length in `getTakeoverDetails`. Then in the loaders code we could just check if `takeoverDetails?.type === 'vimeo'` (or youtube)